fuse/file.go

```diff
@@ -55,11 +56,7 @@ func (fi *File) Attr(ctx context.Context, attr *fuse.Attr) error {
 			attr.Mode = os.ModeSymlink | filePerm
 		}
 	}
-	if fi.hd != nil && fi.hd.writers > 0 {
-		attr.Size = uint64(len(fi.hd.data))
-	} else {
-		attr.Size = info.Size
-	}
+	attr.Size = info.Size
```

Comment:
I'm not entirely sure, but ``info.Size`` might not be correct. When we truncate or extend a file then ``catfs`` still remembers the old size (it gets updated only after Stage(), i.e. on Flush()). We might need to use the limits of the overlay to determine the size at that point. At least that's what I remember, in any case we should check the file size on this test:

```bash
# file should have a size of 1MB.
$ dd if=/dev/zero of=/fuse/file.txt bs=1M count=1
# file size should increase with each write happening.
# (probably easier to test in a unit test where you can
#  control when the file is closed). 
$ dd if=/dev/zero of=/fuse/file.txt bs=1M count=20

```

Reply:
This might also explains why your test produced that much metadata: Every fuse write called ``catfs.Handle.Write()`` and it stages the file to update the file size. That's actually something very dumb to do, and we should retrieve the current files ize from the handle and not from ``catfs`` (where it will be changed on flush).